src/emit.h

```diff
@@ -0,0 +1,71 @@
+#ifndef EMIT_H
+#define EMIT_H
+
+#include "json.h"
+#include "string_id.h"
+#include "field.h"
+
+class emit;
+using emit_id = string_id<emit>;
+
+class emit
+{
+    public:
+        emit() : id_( NULL_ID ) {}
+
+        const emit_id &id() const {
+            return id_;
+        }
+
+        /** When null @ref field is always fd_null */
+        bool is_null() const {
```

Comment:
I'm not familiar with `generic_factory` but will look at it later. The above code example is trivial and can easily wait.